src/engine/engine_wt.c

```diff
@@ -95,48 +159,11 @@ static inline void _ocf_write_wt_submit(struct ocf_request *req)
 			_ocf_write_wt_core_complete);
 }
 
-static void _ocf_write_wt_update_bits(struct ocf_request *req)
-{
-	bool miss = ocf_engine_is_miss(req);
-	bool dirty_any = req->info.dirty_any;
-	bool repart = ocf_engine_needs_repart(req);
-
-	if (!miss && !dirty_any && !repart)
-		return;
-
-	ocf_hb_req_prot_lock_wr(req);
-
-	if (miss) {
-		/* Update valid status bits */
-		ocf_set_valid_map_info(req);
-	}
-
-	if (dirty_any) {
-		/* Writes goes to both cache and core, need to update
-		 * status bits from dirty to clean
-		 */
-		ocf_set_clean_map_info(req);
-	}
-
-	if (repart) {
-		OCF_DEBUG_RQ(req, "Re-Part");
-		/* Probably some cache lines are assigned into wrong
-		 * partition. Need to move it to new one
-		 */
-		ocf_user_part_move(req);
-	}
-
-	ocf_hb_req_prot_unlock_wr(req);
-}
-
 static int _ocf_write_wt_do(struct ocf_request *req)
 {
 	/* Get OCF request - increase reference counter */
 	ocf_req_get(req);
 
```

Comment:
We can detect the case where metadata flush is surely not necessary (!req->info.dirty_any) and set status bits here to avoid extra round trip via I/O queue

Reply:
+ add BUG_ON(!req->info.dirty_any && req->info.flush_metadata) after setting the bits here to make sure our assumption about metadata flush not being necessary holds.